angular-with-directus/how to integrate directus with Angular.md

```diff
@@ -0,0 +1,449 @@
+
+### Configure Directus
+In your Directus project, create a new collection named `pages` - make the Priary ID Field a "Manually Entered String" called `slug`, which will correlate with the URL for the page. For example, `privacy` will later correlate to the page `localhost:4200/privacy`.
+
+Create a text input field called `title` and a text area input field called `content`. In the Access Control settings, give the Public role read access to the new collection.
+
+Create some items in the new collection - [here is some sample data](https://github.com/directus-community/getting-started-demo-data).
+
+### Dynamic Routes in Angular
+Navigate to your project directory in a terminal and generate the page component:
+
+```bash
+ng g c component/page
+```
+Replace the code in the `src/app/component/page/page.component.ts` file with the following code:
+
+```ts
+import {Component, OnInit} from '@angular/core';
+import {directus, Page} from "../../../../directus";
+import {ActivatedRoute} from "@angular/router";
+import {CommonModule} from "@angular/common";
+import {readItem} from "@directus/sdk";
+
+@Component({
+  selector: 'app-page',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './page.component.html',
+  styleUrl: './page.component.css'
+})
+export class PageComponent implements OnInit{
+  page: Page;
+
+  constructor(private route: ActivatedRoute) {
+  }
+
+  ngOnInit(): void {
+    this.route.paramMap.subscribe(params => {
+      const slug = params.get("slug");
+      if (slug){
+        this.getPageBySlug(slug);
+      }
+    })
+  }
+
+  async getPageBySlug(slug: string){
+    //@ts-ignore
+    this.page = await directus
+      .request<Page>(readItem("pages", slug));
+  }
+
+}
+```
+When the component is initialized, the `slug` path parameter is retrieved using `ActivatedRoute` and passed to the `readItem()` method to get a page with that slug.
+
+The retrieved page is stored in the object named `page`. To display the contents of the page, replace the code in the `src/app/component/page/page.component.html` file with the following code:
+
+```html
+<div *ngIf="page">
+  <h1>{{page.title}}</h1>
+  <p>{{page.content}}</p>
+</div>
+```
+### Add Routing for the Pages
+In `src/app/app.routes.ts` add the following route in the `Routes` array:
+
+```ts
+   {path: ':slug', component: PageComponent},
+```
+when the application reloads, go to `http://localhost:4200/privacy` to view the privacy page. Replace the `slug` path parameter with `about` and `conduct` to view the content of about and conduct pages.
+
+## Creating Blog Posts with Directus
+In your Directus project, create a new collection called `authors` with a single text input field called `name`. Add some authors to the collection.
+
+
+Next, create a new collection named `posts` - make the Primary ID Field an "Auto-incremented integer" called `id` which will correlate with the URL for the page. For example, `1` will later correlate to the page `localhost:3000/blog/1`.
```

Comment:
Use a slug with a manually-entered string please, as per the other tutorials.

Reply:
Okay